### PR TITLE
Add "Behind The Scenes" detail

### PR DESCRIPTION
**Pull request by mikenye**
As discussed in Discord: https://discord.com/channels/207038656311984139/816270216489074688/1167614069109444660

### CODE REVIEW

**Comment by mikenye** [comment]
Might want to include some information surrounding ClickHouse?

**Comment by mikenye** [comment]
@bluntelk suggests: we could also add in ingest_tap for seeing how things are working

File: content/behind_the_scenes/index.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+showDate: false
+draft: false
+---
+
+## Overview ##
+
+This article aims to explain how we have re-designed and re-written the backend used to collect, enrich and display the ADSB data our community sends us. 
+
+## Anatomy of a Feeder Connection ##
+When you follow our [docker_planewatch]
```
Comment (sam-ward) [inline content/behind_the_scenes/index.md]: When you run/deploy our [Planewatch docker container/software](https://github.com/plane-watch/docker-plane-watch)

File: content/behind_the_scenes/index.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+![plane.watch feeder connection anatomy](Connection%20Anatomy.drawio.png)
+
+1. [pw-feeder][pw-feeder] opens two connections to [Bordercontrol](#bordercontrol) - one connection for [BEAST][beast protocol], and another for [MLAT][mlat-client].
+2. During connection establishment, [Bordercontrol](#bordercontrol) checks the validity of the feeder's API key. If invalid, the connection is dropped. If valid, the connection is allowed, and [bordercontrol](#bordercontrol) starts a "feed-in" container for the client.
+3. Bordercontrol proxies [BEAST][beast protocol] traffic to the feed-in container.
+4. Bordercontrol proxies [MLAT][mlat-client] traffic to an [mlat_server][mlat-server] instance running on the regional multiplexer.
+5. For the "new" environment, [pw_ingest](#pw_ingest) running within the feed-in container decodes the [BEAST][beast protocol] data, and publishes the data as a message onto the NATS message bus. The data is processed through the [pw-pipeline][pw-pipeline].
+6. For the "legacy" environment, the [BEAST][beast protocol] data for each region is multiplexed in each regional multiplexer. Virtual Radar Server then consumes this data for the legacy front-end.
```
Comment (sam-ward) [inline content/behind_the_scenes/index.md]: Is this relevant??, given the page is about the new architecture.  Maybe move that portion to a "legacy" setup page

File: content/behind_the_scenes/index.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+## Components ##
+
+### ADS-B Traffic Control (ATC) ###
+
+[ATC][atc] is [plane.watch][plane.watch]'s feeder portal. It provides:
+
+* User sign-up and account management
+* Feeder creation and management
+* Databases containing aircraft types, airports, routes, operators and waypoints
```
Comment (sam-ward) [inline content/behind_the_scenes/index.md]: * A database containing...

File: content/behind_the_scenes/index.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+[ATC][atc] provides both public and private APIs. The public API is used by [pw-feeder](#pw-feeder) to report the feeder's status back to the client.
+
+The private API is used to:
+
+* Authenticate feeders upon connection to [bordercontrol](#bordercontrol)
+* Provide aircraft types, airports, routes, operators and waypoints to the [Enrichment Centre](#pw-enrichment)
+
+### pw-feeder ###
+
+[pw-feeder][pw-feeder], simply put, it is a [BEAST][beast protocol] and [MLAT][mlat] protocol specific [stunnel][stunnel] client, that securely proxies data from the client to [plane.watch][plane.watch]'s servers (specifically [bordercontrol](#bordercontrol) - see below).
```
Comment (sam-ward) [inline content/behind_the_scenes/index.md]: [pw-feeder][pw-feeder] is, simply put, a [BEAST][beast protocol] and [MLAT][mlat] protocol specific [stunnel][stunnel] client, that securely proxies data from the client to the [plane.watch][plane.watch] servers (specifically [bordercontrol](#bordercontrol) - see below).

File: content/behind_the_scenes/index.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+[pw-feeder][pw-feeder] runs on the feeder client computer, and receives [BEAST protocol][beast protocol] data from software such as [dump1090][dump1090] and [readsb][readsb], or from hardware devices such as the [Jetvision Radarcape][radarcape]. It also communicates with [mlat-client][mlat-client] software running on the client, and mlat-server software running within [plane.watch][plane.watch].
```
Comment (sam-ward) [inline content/behind_the_scenes/index.md]: I'm nitpicking now but in the diagram you call it "feeder computer", not "feeder client computer".

File: content/behind_the_scenes/index.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+## Overview ##
+
+This article aims to explain how we have re-designed and re-written the backend used to collect, enrich and display the ADSB data our community sends us. 
+
+## Anatomy of a Feeder Connection ##
+When you follow our [docker_planewatch] 
+![plane.watch feeder connection anatomy](Connection%20Anatomy.drawio.png)
+
+1. [pw-feeder][pw-feeder] opens two connections to [Bordercontrol](#bordercontrol) - one connection for [BEAST][beast protocol], and another for [MLAT][mlat-client].
+2. During connection establishment, [Bordercontrol](#bordercontrol) checks the validity of the feeder's API key. If invalid, the connection is dropped. If valid, the connection is allowed, and [bordercontrol](#bordercontrol) starts a "feed-in" container for the client.
```
Comment (sam-ward) [inline content/behind_the_scenes/index.md]: This is a bit of duplication with the bordercontrol section below. Not sure if it should be slimmed down/simplified here and then expanded on below.  Probably to detailed here for the first overview section.

File: content/behind_the_scenes/index.md (hunk 2 above)
Comment (sam-ward) [inline content/behind_the_scenes/index.md]: Add a 7. Data is fetched (via a websocket)by the web interface for displaying to the end users....

You dont actually mention the front end anywhere until the pw-router section.

File: content/behind_the_scenes/index.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+### pw-enrichment ###
+
+**pw-enrichment**, also known as EC (enrichment centre), consumes decoded ADS-B frames from the [NATS][nats] message queue and adds additional information relating to the aircraft, route and other metadata. The EC maintains an internal memory-cache so as to not overwhelm the ATC database, given the message rate can be in the thousands of messages a second.
+
+### pw_router ###
+
+[**pw_router**][pw-pipeline] takes enriched data and reduces it down to significant events, thus cutting the message rate significantly. The router also publishes two feeds to the NATS bus - a high resolution and a low resolution - these are used in the UI to not overwhelm the browser with messages.
+
+Router also connects to an instance of [ClickHouse], a highly scalable columnar database designed for storing timeseries data - perfect for ADSB. Both high and low resolution feeds are stored to Clickhouse for later analysis and for drawing the historical paths in the UI.
```
Comment (sam-ward) [inline content/behind_the_scenes/index.md]: What about raw data. is that discarded or also stored in clickhouse?  Should clickhouse be added to the diagram at the top of the page? If nothing else only for context.

File: content/behind_the_scenes/index.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+### pw_ws_broker ###
+
+[**pw_ws_broker**][pw-pipeline] is a horizontally scalable service that provides a [WebSocket][websocket] interface to connect the public web interface with the backend pipeline. When a user loads the public website, a websocket is dialed up to this service to stream ADSB updates from the pipeline directly to the user.
```
Comment (sam-ward) [inline content/behind_the_scenes/index.md]: The terminolgy between UI and website is mixed throughout the document. ideally would be consistent.  Makes it seem like there is 2 different user interfaces.

"a websocket is s/dialed up/connected to this service".  

Does it stream ADSB? Or does it stream the high/low resolution feeds?  You talk about ADSB at the start, but the rest of the document just deals with messages.

File: content/behind_the_scenes/index.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+The broker also provides streaming APIs for querying ATC for more detailed route and airframe information as well as querying ClickHouse for historical track information.
```
Comment (sam-ward) [inline content/behind_the_scenes/index.md]: The broker also provides streaming APIs for querying ATC for more detailed route and airframe information, as well as querying ClickHouse for historical track information.

**Comment by tardoe** [comment]
> @bluntelk suggests: we could also add in ingest_tap for seeing how things are working

I'd make this a whole other post honestly - gotta keep some cool stuff up our sleeves!

File: content/behind_the_scenes/index.md (hunk 1 above)
Comment (tardoe) [reply content/behind_the_scenes/index.md]: I'll update this to point to the getting started guide on this site which also links to that GH page.

File: content/behind_the_scenes/index.md (hunk 2 above)
Comment (tardoe) [reply content/behind_the_scenes/index.md]: Fixed, good call.

Comment (tardoe) [reply content/behind_the_scenes/index.md]: Added an extra point.

Comment (tardoe) [reply content/behind_the_scenes/index.md]: @mikenye can you update the diagram to remove the legacy stuff and add a point 6 for the frontend part of it?

File: content/behind_the_scenes/index.md (hunk 7 above)
Comment (tardoe) [reply content/behind_the_scenes/index.md]: We don't store raw data, just parsed frames.

File: content/behind_the_scenes/index.md (hunk 8 above)
Comment (tardoe) [reply content/behind_the_scenes/index.md]: Fair point, I'll clarify.

File: content/behind_the_scenes/index.md (hunk 7 above)
Comment (sam-ward) [reply content/behind_the_scenes/index.md]: Maybe worth just stating that raw frames are discard after processing, just for completeness.

File: content/behind_the_scenes/index.md (hunk 2 above)
Comment (mikenye) [reply content/behind_the_scenes/index.md]: @tardoe done. Updated the list slightly, would suggest re-review. :-)

**Review by bluntelk — APPROVED** [review]

**Review by shaneshort — APPROVED** [review]

**Pull request merged**